From d7b16cce28a629e1785aaa788cc53d858546c3a1 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Tue, 6 Nov 2012 16:04:52 +0800
Subject: [PATCH 364/635] ENGR00222067 wifi: fix parallel compile error

1.Like before, delete record date when generate compat_autoconf.h.
  this can avoid git dirty.
2.Like $CURDIR, $PWD variable, kernel also define them. For we add wifi
  driver out of kernel and also use $CURDIR, $PWD. so when make -jN in
  top make, kernel and wifi driver may compile in two different jobs.
  if wifi driver firstly begin, then the $CURDIR,$PWD value will be overwriten
  by the kernel.so outside of kernel in android, avoid using thus variables.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 wlan/atheros/compat-wireless/Makefile                  | 18 +++++++++---------
 .../compat-wireless/include/linux/compat_autoconf.h    |  1 -
 .../compat-wireless/scripts/gen-compat-autoconf.sh     |  3 +--
 3 files changed, 10 insertions(+), 12 deletions(-)

diff --git a/wlan/atheros/compat-wireless/Makefile b/wlan/atheros/compat-wireless/Makefile
index d3256c1..47af45f 100644
--- a/wlan/atheros/compat-wireless/Makefile
+++ b/wlan/atheros/compat-wireless/Makefile
@@ -38,7 +38,7 @@ endif
 
 else
 
-export CURDIR := $(shell pwd)
+export COMPAT_ROOT := $(shell pwd)
 CFLAGS += \
         -DCOMPAT_BASE_TREE="\"$(shell cat compat_base_tree)\"" \
         -DCOMPAT_BASE_TREE_VERSION="\"$(shell cat compat_base_tree_version)\"" \
@@ -50,22 +50,22 @@ CFLAGS += \
 export COMPAT_CONFIG=config.mk
 export CONFIG_CHECK=.$(COMPAT_CONFIG)_md5sum.txt
 export COMPAT_AUTOCONF=include/linux/compat_autoconf.h
-export CREL=$(shell cat $(CURDIR)/compat_version)
+export CREL=$(shell cat $(COMPAT_ROOT)/compat_version)
 export CREL_PRE:=.compat_autoconf_
 export CREL_CHECK:=$(CREL_PRE)$(CREL)
 
-include $(CURDIR)/$(COMPAT_CONFIG)
+include $(COMPAT_ROOT)/$(COMPAT_CONFIG)
 
 all: modules
 
 modules: $(CREL_CHECK)
 	@./scripts/check_config.sh
-	$(MAKE) -C $(KLIB_BUILD) M=$(CURDIR) modules
+	$(MAKE) -C $(KLIB_BUILD) M=$(COMPAT_ROOT) modules
 	@touch $@
 
 bt: $(CREL_CHECK)
 	@./scripts/check_config.sh
-	$(MAKE) -C $(KLIB_BUILD) M=$(CURDIR) BT=TRUE modules
+	$(MAKE) -C $(KLIB_BUILD) M=$(COMPAT_ROOT) BT=TRUE modules
 	@touch $@
 
 # With the above and this we make sure we generate a new compat autoconf per
@@ -81,7 +81,7 @@ $(CREL_CHECK):
 btinstall: btuninstall bt-install-modules
 
 bt-install-modules: bt
-	$(MAKE) -C $(KLIB_BUILD) M=$(CURDIR) $(KMODDIR_ARG) $(KMODPATH_ARG) BT=TRUE \
+	$(MAKE) -C $(KLIB_BUILD) M=$(COMPAT_ROOT) $(KMODDIR_ARG) $(KMODPATH_ARG) BT=TRUE \
 		modules_install
 	@/sbin/depmod -ae
 	@echo
@@ -148,13 +148,13 @@ btuninstall:
 	@echo
 
 btclean:
-	make -C /lib/modules/$(shell uname -r)/build M=$(CURDIR) BT=TRUE clean
+	make -C /lib/modules/$(shell uname -r)/build M=$(COMPAT_ROOT) BT=TRUE clean
 	@rm -f $(CREL_PRE)*
 
 install: uninstall install-modules install-scripts
 
 install-modules: modules
-	$(MAKE) -C $(KLIB_BUILD) M=$(CURDIR) $(KMODDIR_ARG) $(KMODPATH_ARG) \
+	$(MAKE) -C $(KLIB_BUILD) M=$(COMPAT_ROOT) $(KMODDIR_ARG) $(KMODPATH_ARG) \
 		modules_install
 
 install-scripts:
@@ -439,7 +439,7 @@ uninstall:
 
 clean:
 	@if [ -d net -a -d $(KLIB_BUILD) ]; then \
-		$(MAKE) -C $(KLIB_BUILD) M=$(CURDIR) clean ;\
+		$(MAKE) -C $(KLIB_BUILD) M=$(COMPAT_ROOT) clean ;\
 	fi
 	@rm -f $(CREL_PRE)*
 unload:
diff --git a/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h b/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
index 740c5c2..70d5432 100644
--- a/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
+++ b/wlan/atheros/compat-wireless/include/linux/compat_autoconf.h
@@ -2,7 +2,6 @@
 #define COMPAT_AUTOCONF_INCLUDED
 /*
  * Automatically generated C config: don't edit
- * Fri Nov  2 10:11:00 CST 2012 
  * compat-wireless-2.6: patch-10-27-gd46416c-p
  * linux-2.6: branching-msm-3.2.3-44-g65b285b
  */
diff --git a/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh b/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh
index 28ed25a..b577d11 100755
--- a/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh
+++ b/wlan/atheros/compat-wireless/scripts/gen-compat-autoconf.sh
@@ -1,6 +1,7 @@
 #!/bin/bash
 # 
 # Copyright 2007	Luis R. Rodriguez <mcgrof@winlab.rutgers.edu>
+# Copyright (C) 2012 Freescale Semiconductor, Inc.
 #
 # Use this to parse a small .config equivalent looking file to generate
 # our own autoconf.h. This file has defines for each config option
@@ -35,7 +36,6 @@ fi
 
 CREL=$(cat $COMPAT_RELEASE | tail -1)
 KREL=$(cat $KERNEL_RELEASE | tail -1)
-DATE=$(date)
 
 # Defines a CONFIG_ option if not defined yet, this helps respect
 # linux/autoconf.h 
@@ -133,7 +133,6 @@ cat <<EOF
 #define COMPAT_AUTOCONF_INCLUDED
 /*
  * Automatically generated C config: don't edit
- * $DATE 
  * compat-wireless-2.6: $CREL
  * linux-2.6: $KREL
  */
-- 
1.8.0

